src/engine/openrpc/schemas/payload.yaml

```
@@ -0,0 +1,202 @@
+PayloadStatusV1:
+  title: Payload status object V1
+  type: object
+  required:
+    - status
+  properties:
+    status:
+      title: Payload validation status
+      type: string
+      enum:
+        - VALID
+        - INVALID
+        - SYNCING
+        - ACCEPTED
+        - INVALID_BLOCK_HASH
+    latestValidHash:
+      title: The hash of the most recent valid block
+      $ref: '#/components/schemas/hash32'
+    validationError:
+      title: Validation error message
+      type: string
+RestrictedPayloadStatusV1:
+  $ref: '#/components/schemas/PayloadStatusV1'
+  properties:
+    status:
+      $ref: '#/components/schemas/PayloadStatusV1/properties/status'
+      description: Set of possible values is restricted to VALID, INVALID, SYNCING
+      enum:
+        - VALID
+        - INVALID
+        - SYNCING        
+    latestValidHash:
+      $ref: '#/components/schemas/PayloadStatusV1/properties/latestValidHash'
+    validationError:
+      $ref: '#/components/schemas/PayloadStatusV1/properties/validationError'
+PayloadStatusV2:
+  $ref: '#/components/schemas/PayloadStatusV1'
+  title: Payload status object V2
+  properties:
+    status:
+      $ref: '#/components/schemas/PayloadStatusV1/properties/status'
+      enum:
+        - VALID
+        - INVALID
+        - SYNCING
+        - ACCEPTED        
+    latestValidHash:
+      $ref: '#/components/schemas/PayloadStatusV1/properties/latestValidHash'
+    validationError:
+      $ref: '#/components/schemas/PayloadStatusV1/properties/validationError'
+ExecutionPayloadV1:
+  title: Execution payload object V1
+  type: object
+  required:
+    - parentHash
+    - feeRecipient
+    - stateRoot
+    - receiptsRoot
+    - logsBloom
+    - prevRandao
+    - blockNumber
+    - gasLimit
+    - gasUsed
+    - timestamp
+    - extraData
+    - baseFeePerGas
+    - blockHash
+    - transactions
+  properties:
+    parentHash:
+      title: Parent block hash
+      $ref: '#/components/schemas/hash32'
+    feeRecipient:
+      title: Recipient of transaction priority fees
+      $ref: '#/components/schemas/address'
+    stateRoot:
+      title: State root
+      $ref: '#/components/schemas/hash32'
+    receiptsRoot:
+      title: Receipts root
+      $ref: '#/components/schemas/hash32'
+    logsBloom:
+      title: Bloom filter
+      $ref: '#/components/schemas/bytes256'
+    prevRandao:
+      title: Previous randao value
+      $ref: '#/components/schemas/bytes'
+    blockNumber:
+      title: Block number
+      $ref: '#/components/schemas/uint64'
+    gasLimit:
+      title: Gas limit
+      $ref: '#/components/schemas/uint64'
+    gasUsed:
+      title: Gas used
+      $ref: '#/components/schemas/uint64'
+    timestamp:
+      title: Timestamp
+      $ref: '#/components/schemas/uint64'
+    extraData:
+      title: Extra data
+      $ref: '#/components/schemas/bytes'
```

```suggestion
      $ref: '#/components/schemas/bytes32'
```
Need to double check, but I think this works for length 0 to 32

---

Hmm, I believe `bytes32` is a vector of 32 bytes while this field has variable length between 0 and 32.

---

Ah right, okay so we do have `bytesMax32` so I think that is the correct type to use: https://github.com/ethereum/execution-apis/blob/517cdbcf7f19b25aff3120fa4f567637a26fcbcd/src/schemas/base-types.yaml#L18